TruffleString-Experiment.md

```diff
@@ -10,11 +10,8 @@
 
 TruffleSqueak is a Squeak implementation on the GraalVM Truffle framework [6]. It aims to leverage the performance and language interoperability benefits of Truffle while maintaining compatibility with Squeak's features. We worked with TruffleSqueak 24.2.2 [9]. Each Squeak object is represented as a Java object.
 
-This experiment focuses on the data representation of Smalltalk objects in Java. We differentiate between primitives (i.e. `ClassObject`) and non-primitives (i.e. `ByteString`, `ByteArray`, etc.). Primitive classes have a direct mapping to Java classes.
-
-![Primitive Object Representation, Example: `ClassObject`](img/graal-fmc-primitive.drawio.png)
-
-Non-primitive classes are represented by Java objects of the class `NativeObject` that encapsulate the data. Each `NativeObject` instance contains a `storage` and `class` object. The `storage` contains the actual data, while the `class` object provides metadata about the object type. Different non-primitive classes have different storage types, such as `WideString` uses `int[]` and `ByteString` uses `byte[]`.
+This experiment focuses on how Smalltalk objects are represented in Java, particularly the representation of indexable Smalltalk objects like strings, arrays, and byte arrays.
```

Comment:
In the sentence prior we write that "Each Squeak object is represented as a Java object." and now we talk about "Smalltalk objects" or even "indexable Smalltalk objects" (btw: I really like the "indexable" specification 😄).
Should we stick to a consistent terminalogy and always talk about "Squeak/Smalltalk objects"? 🤔

Reply:
Yes! Fabio used "Squeak object", so I'd stick with it